components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -310,8 +307,20 @@ export function ConversationContextProvider(
       })
   }, [context.conversationUuid, context.faviconCacheKey])
 
-  const actionList = useActionMenu(context.inputText, aiChatContext.allActions
-  )
+  // Update the location when the visible conversation changes
+  const isVisible = useIsConversationVisible(context.conversationUuid)
+  React.useEffect(() => {
+    if (!isVisible) return
+    if (selectedConversationId === tabAssociatedChatId) return
+    if (context.conversationUuid === selectedConversationId) return
+    updateSelectedConversationId(context.conversationUuid)
+  }, [isVisible, updateSelectedConversationId])
+
+  React.useEffect(() => {
+    setPartialContext({ isTabAssociated: selectedConversationId === tabAssociatedChatId })
```

Comment:
Would having `isTabAssociated` be provided from `ActiveChatContext` avoid an extra round of state setting? Does it matter?

Reply:
Ah yup, that was a hangover from before the ActiveChatContext was a context :smile: